### PR TITLE
Improve workflow to release images 

### PR DESCRIPTION
**Pull request by potiuk**
Adding debugging and explaining that we should use the
stable branch to release the images

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/README_RELEASE_AIRFLOW.md [hunk 1]
```diff
@@ -664,7 +662,9 @@ Building the image is triggered by running the
 When you trigger it you need to pass:
 
 * Airflow Version
-* Optional "true" in skip latest field if you do not want to retag the latest image
+* Optional "true" in skip latest field if you do not want to re-tag the latest image
```
Comment (pierrejeambrun) [inline dev/README_RELEASE_AIRFLOW.md:668]: Maybe we can add something like "`rc` suffix and other pre-release version will automatically set the `skip latest` to true".

(Or a different wording, but I remembered wondering if I should skip or not for rc, and had to check the code)

**Review by pierrejeambrun — APPROVED** [review]
Nice thanks

File: dev/README_RELEASE_AIRFLOW.md [hunk 2]
```diff
@@ -313,10 +313,8 @@ to have an environment prepared to build multi-platform images. You can achieve
 Building the image is triggered by running the
 [Release PROD Images](https://github.com/apache/airflow/actions/workflows/release_dockerhub_image.yml) workflow.
 
-When you trigger it you need to pass:
-
-* Airflow Version (including the right rc suffix)
-* Optional "true" in the "Skip latest:" field if you do not want to re-tag the latest image
+When you trigger it you need to pass Airflow Version (including the right rc suffix). Make sure to use the
+``v2-*-stable`` branch for the workflow.
```
Comment (potiuk) [inline dev/README_RELEASE_AIRFLOW.md:317]:
```suggestion
``v2-*-stable`` branch for the workflow.

You can leave the "skip latest" field empty.

```

File: dev/README_RELEASE_AIRFLOW.md (hunk 1 above)
Comment (potiuk) [reply dev/README_RELEASE_AIRFLOW.md:668]: Good idea. Added.

**Pull request merged**